flaml/autogen/agent/agent.py

```diff
@@ -30,22 +31,73 @@ def _remember(self, memory):
         """Remember something."""
         self._memory.append(memory)
 
-    def _send(self, message, recipient):
+    def _send(self, message: Union[Dict, str], recipient):
         """Send a message to another agent."""
-        self._conversations[recipient.name].append({"content": message, "role": "assistant"})
+        if type(message) is str:
+            oai_message = {"content": message, "role": "assistant"}
+        else:
+            # create openai message to be appended to the conversation
+            oai_message = {k: message[k] for k in ("content", "function_call", "name") if k in message}
+            # When the agent composes and sends the message, the role of the message is "assistant". The role of 'function' will remain unchanged.
+            oai_message["role"] = "function" if message.get("role") == "function" else "assistant"
+        self._oai_conversations[recipient.name].append(oai_message)
+
         recipient.receive(message, self)
 
-    def _receive(self, message, sender):
-        """Receive a message from another agent."""
-        print("\n", "-" * 80, "\n", flush=True)
-        print(sender.name, "(to", f"{self.name}):", flush=True)
-        print(message, flush=True)
-        self._conversations[sender.name].append({"content": message, "role": "user"})
+    def _receive(self, message: Union[Dict, str], sender):
+        """Receive a message from another agent.
+
+        Args:
+            message (dict or str): message from the sender. If the type is dict, it can contain at most 4 fields:
+                1. "content": content of the message, can be None.
```

Comment:
```suggestion
                1. "content" (required): content of the message, can be None.
```

Reply:
This might not be required. The message can only contain a 'function_call' field